crates/types/src/simple_vote.rs

```diff
@@ -0,0 +1,225 @@
+//! Implementations of the simple vote types.
+
+use std::{fmt::Debug, hash::Hash, marker::PhantomData};
+
+use commit::{Commitment, Committable};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    traits::{
+        election::Membership,
+        node_implementation::NodeType,
+        signature_key::{EncodedPublicKey, EncodedSignature, SignatureKey},
+    },
+    vote2::{HasViewNumber, Vote2},
+};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Hash, Eq)]
+/// Data used for a yes vote.
+pub struct YesData<LEAF: Committable> {
+    /// Commitment to the leaf
+    pub leaf_commit: Commitment<LEAF>,
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Hash, Eq)]
+/// Data used for a DA vote.
+pub struct DAData<BLOCK: Committable> {
```

Comment:
Is the `BLOCK` a block payload, or can it be something else? If it's a block payload, since I'm refactoring the payload to a struct, we could then remove the `BLOCK` generic. Before the refactor, I think we could rename `block_commit` and `BLOCK` to `payload_commit` and `PAYLOAD`.

Reply:
I honestly hadn't really thought it through.  It's really the thing that the DA committee receives and votes on.  I think this should be the block payload, so I'll rename to that for now unless you have a different thought.

Reply:
Sounds good!